src/javascript/app_2/Modules/Trading/Containers/purchase.jsx

```diff
@@ -33,37 +30,22 @@ const Purchase = ({
     Object.keys(trade_types).map((type, idx) => {
         const info        = proposal_info[type] || {};
         const is_disabled = !is_purchase_enabled || !is_trade_enabled || !info.id || !is_client_allowed_to_visit;
-        const is_purchase_error = (!isEmptyObject(purchase_info) && purchase_info.echo_req.buy === info.id);
         const is_high_low = /high_low/.test(contract_type.toLowerCase());
-
+        const is_loading  = !info.has_error && !info.id;
         const purchase_button = (
-            <Button
-                is_disabled={is_contract_mode || is_disabled}
-                id={`purchase_${type}`}
-                className='btn--primary btn-purchase'
-                has_effect
-                onClick={() => {
-                    onClickPurchase(info.id, info.stake, type);
-                }}
-            >
-                <React.Fragment>
-                    <div className='btn-purchase__effect-main' />
-                    <div className='btn-purchase__effect-detail' />
-                    <div className='btn-purchase__content'>
-                        <div className='btn-purchase__trade-type'>
-                            <IconTradeType type={!is_high_low ? type.toLowerCase() : `${type.toLowerCase()}_barrier`} className='btn-purchase__trade-type-icon' />
-                            <span className='btn-purchase__trade-type-text'>{localize('[_1]', trade_types[type])}</span>
-                        </div>
-                    </div>
-                    <div className='btn-purchase__info'>
-                        <div className='btn-purchase__return'>{is_disabled ? '---,-' : info.returns}</div>
-                        <div className='btn-purchase__profit'>
-                            {is_disabled ? '--,--' : <Money amount={info.profit} currency={currency} className='btn-purchase__currency' />}
-                        </div>
-                    </div>
-                </React.Fragment>
-            </Button>
+            <PurchaseButton
+                currency={currency}
+                info={info}
+                is_contract_mode={is_contract_mode}
+                is_disabled={is_disabled}
+                is_high_low={is_high_low}
+                is_loading={is_loading}
+                onClickPurchase={onClickPurchase}
+                trade_types={trade_types}
+                type={type}
+            />
         );
+        const is_purchase_error = (!isEmptyObject(purchase_info) && purchase_info.echo_req.buy === info.id);
```

Comment:
can this be above `purchase button`

Reply:
This should be under purchase button since it isn't used by it. I think it is good to just create a new variables/object/functions right when you need to use it.